tests/core/sequence/test_sequence_manager.py

```diff
@@ -189,25 +192,111 @@ def test_get_all_on_multiple_versions_environment():
 
 
 def test_is_submittable():
-    dn = InMemoryDataNode("dn", Scope.SCENARIO, properties={"default_data": 10})
-    task = Task("task", {}, print, [dn])
-    scenario = Scenario("scenario", {task}, {}, set())
+    task_id = "TASK_task_id"
+    scenario_id = "SCENARIO_scenario_id"
+    dn_1 = PickleDataNode("dn_1", Scope.SCENARIO, parent_ids={task_id, scenario_id}, properties={"default_data": 10})
+    dn_2 = PickleDataNode("dn_2", Scope.SCENARIO, parent_ids={task_id, scenario_id}, properties={"default_data": 10})
+    task = Task("task", {}, print, [dn_1, dn_2], id=task_id, parent_ids={scenario_id})
+    scenario = Scenario("scenario", {task}, {}, set(), scenario_id=scenario_id)
+    _DataManager._set(dn_1)
+    _DataManager._set(dn_2)
+    _TaskManager._set(task)
     _ScenarioManager._set(scenario)
 
+    dn_1 = scenario.dn_1
```

Comment:
The new test cases should be moved to a dedicated file for ReadyToRunProperty class.